src/js/TextCopyHelper.js

```diff
@@ -1,60 +1,99 @@
-/*global require, module*/
-"use strict";
+/**global require, module*/
+'use strict';
 
 var DomDelegate = require('ftdomdelegate');
-var ToolTip = require('./ToolTip');
-var shareDom = require('./shareDom');
+var Tooltip = require('./Tooltip');
 
+/**
+ * Gets the width of a text by using a <canvas> element
+ *
+ * @param {string} text - The text to measure
+ * @param {HTMLElement} refEl - The reference element where the text will be to get the font properties
+ *
+ * @returns {number}
+ */
+function getPixelWidthOfText(text, refEl) {
+	var c = document.createElement("canvas");
+
+	if (c.getContext && window.getComputedStyle) {
+		var ctx = c.getContext("2d");
+		var refElStyle = window.getComputedStyle(refEl);
+		ctx.font = refElStyle.getPropertyValue('font-style') + " " + refElStyle.getPropertyValue('font-size') + " " + refElStyle.getPropertyValue('font-family');
+		ctx.fillText(text, 10, 100);
+		return ctx.measureText(text).width;
+	}
+
+	return -1;
+}
+
+/**
+ * @class TextCopyHelper
+ *
+ * @param {Object} config
+ * @param {string} config.text - Value of the url input element
+ * @param {string} config.message - Tooltip text
+ * @param {HTMLElement} config.parentEl - Parent element
+ * @param {function} config.onDestroy - Optional, callback that will be ran on {@link destroy}
+ * @param {function} config.onCopy - Optional, callback that will be ran when the 'copy' event is triggered
+ */
 function TextCopyHelper(config) {
```

Comment:
I say we go with whatever is easiest to understand: fail hard and early if the programmer makes a mistake, help the reader understand what's going on - computers will do whatever.   I'm not strongly opinionated unless the API makes absolutely no sense.

My _only_ concern is a method that has potentially ambiguous primitive values with no indicator as to what they're for unless you read the docs:

```JS
myFunction("Get Help", "More help on this subject", someElement);
```

Using an API like this I'd tend to write:

```JS
var urlValueAttribute = "Get Help";
var toolTip =  "More help on this subject";
myFunction(urlValueAttribute, toolTip, someElement);
```

It will become a non-issue with ES6, but this isn't ES6 :p.